js/interface.js

```diff
@@ -285,19 +285,91 @@ function getFolderContentsById(id, type, isSearchNav) {
   });
 }
 
+function restoreParentFolder(options) {
+  Fliplet.API.request(options.request)
+    .then(function(result) {
+      if (result) {
+        showSpinner(false);
+  
+        $('[data-browse-trash]').click();
```

Comment:
Instead of using `.click()` to trigger an action by proxy, find out the click is meant to do and run that code directly. You may need to wrap up the operations into a function.

By avoiding proxy, we can avoid the complexity of chained events, which means we can effectively use `event.stopPropagation()` etc. when required.